keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt

```diff
@@ -0,0 +1,68 @@
+package com.netflix.spinnaker.keel.services
+
+import com.netflix.spinnaker.igor.BuildService
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactMetadata
+import com.netflix.spinnaker.keel.api.artifacts.BuildMetadata
+import com.netflix.spinnaker.keel.api.artifacts.GitMetadata
+import com.netflix.spinnaker.model.Build
+import io.github.resilience4j.retry.annotation.Retry
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import retrofit2.HttpException
+
+/**
+ * Provides functionality to convert build metadata, which is coming from internal service, to artifact metadata (via igor).
+ */
+@Component
+class ArtifactMetadataService(
+  private val buildService: BuildService
+) {
+
+  @Retry(name = "getArtifactMetadata", fallbackMethod = "getFallback")
```

Comment:
@luispollo need your input here regarding the `retry` -- I've configured it in `keel.yml` but haven't tested it yet. let me know if you think something is missing.

Reply:
@luispollo introduced it to me! it supposed to save you a bunch of code lines

Reply:
@gal-yardeni This should be relatively straightforward to test if you configure your igor endpoint to point to a mock server that returns error codes. There's examples of using `MockWebServer` in the tests (see [here](https://github.com/spinnaker/keel/blob/master/keel-web/src/test/kotlin/com/netflix/spinnaker/keel/integration/AuthPropagationTests.kt)). The only thing that stands out to me in the annotation is that you've configured your `fallbackMethod` to a method named `getFallback` but I don't see that method in your class...

About the configuration, you mentioned you configured retry properties in `keel.yml`. I assume you meant our internal config file. It would be good to add that to https://github.com/spinnaker/keel/blob/master/keel-web/config/keel.yml so that it can be reviewed and used by OSS deployments.

Reply:
added the fallback method, let me know if it makes sense.